mysql/mysqldump.go

```diff
@@ -368,10 +408,22 @@ func updateColsByOption(conv *internal.Conv, tableName string, col *ast.ColumnDe
 		case ast.ColumnOptionCheck:
 			column.Ignored.Check = true
 		case ast.ColumnOptionReference:
-			column.Ignored.ForeignKey = true
+			column := col.Name.String()
+			referTable, err := getTableName(elem.Refer.Table)
+			if err != nil {
+				conv.Unexpected(err.Error())
+				continue
+			}
+			referColumn := elem.Refer.IndexPartSpecifications[0].Column.Name.String()
+			fkey := schema.ForeignKey{Columns: []string{column},
```

Comment:
Just wanted to double check: when we get foreign key constraints as part of a column definition, they are always unnamed -- is that correct?
Hence leaving Name unset is the right thing to do.

Reply:
Yes Nevin, we don't get name for column level constraints

Reply:
Maybe worth a comment? How about something like:

// Note that foreign key constraints that are part of a column definition
// have no name, so we leave fkey.Name as the empty string.

Reply:
added comment